walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -34,23 +38,73 @@
     }
 
     // Seperate relevant TimeRanges from events, put into ArrayList and sort by ascending meeting start time
+    // Optional attendees are handled by running two sets of queries in parallel: a query if we are including optional attendees,
+    // and a query if we are only considering required attendees. If the optional attendee query is empty, then that means that 
+    // there are no valid times if we include optional attendees, and optional attendees are ignored.
     List<TimeRange> attendedMeetings = new ArrayList<>();
+    List<TimeRange> allAttendedMeetings = new ArrayList<>();
+
+    // Attendees is only required attendees, allAttendees includes optional attendees
+    Set<String> requiredAttendees = new HashSet<>(request.getAttendees());
```

Comment:
You cannot take out requiredAttendees because you eventually changes them bellow (retainAll). That reminds me of a lesson that you want to make objects immutable whenever possible. Can you move the requiredAttendees back in and make optionalAttendees ImmutableSet?
Also try to find out other immutable containers throughout the code and change them to immutable.